src/redisearch_rs/CMakeLists.txt

```diff
@@ -0,0 +1,103 @@
+cmake_minimum_required(VERSION 3.13)
+
+# Find Rust and Cargo
+find_program(CARGO_EXECUTABLE cargo REQUIRED)
+
+if(NOT CARGO_EXECUTABLE)
+    message(FATAL_ERROR "Cargo not found. Please install Rust and Cargo.")
+endif()
+
+# Determine Rust profile based on build configuration
+set(RUST_PROFILE "release")
+if(DEFINED BUILD_SEARCH_UNIT_TESTS AND BUILD_SEARCH_UNIT_TESTS)
+    if(CMAKE_BUILD_TYPE STREQUAL "Debug" OR DEFINED SAN OR DEFINED COV)
+        set(RUST_PROFILE "dev")
+    else()
+        set(RUST_PROFILE "optimised_test")
+    endif()
+else()
+    # For non-test builds, only use dev profile for Debug builds (not SAN/COV)
+    if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+        set(RUST_PROFILE "dev")
+    endif()
```

Comment:
This should follow the same pattern of `RUST_DYN_CRT`: allow the invoker to set the profile, but have fallback logic if it isn't set.
We should also remove the corresponding profile selection logic from `build.sh` and just forward `RUST_PROFILE` if it was set by the user.

Reply:
no, for that project is not relevant

Reply:
Then let's keep the profile selection logic in `build.sh` and simplify this down to "either propagate or fallback to `release`".